engine/check.py

```diff
@@ -3,52 +3,71 @@
 
 This module defines a CLI to compare two data files (reference and current)
 against specified tolerances.
-It utilizes utility functions for testing statistical files with tolerances and
-computing divergence between DataFrames.
+It utilizes utility functions for testing statistical and fof files with
+tolerances and computing divergence between DataFrames.
 """
 
 import sys
 
 import click
 
-from util.click_util import cli_help
-from util.dataframe_ops import check_stats_file_with_tolerances, compute_division
+from util.click_util import CommaSeparatedStrings, cli_help
+from util.dataframe_ops import check_file_with_tolerances, compute_division
+from util.fof_utils import expand_zip
 from util.log_handler import logger
 
 
 @click.command()
 @click.option(
-    "--input-file-ref",
-    help=cli_help["input_file_ref"],
+    "--reference-files",
+    type=CommaSeparatedStrings(),
+    help=cli_help["reference_files"],
+    default=None,
 )
 @click.option(
-    "--input-file-cur",
-    help=cli_help["input_file_cur"],
+    "--current-files",
+    type=CommaSeparatedStrings(),
+    help=cli_help["current_files"],
+    default=None,
 )
 @click.option(
-    "--tolerance-file-name",
-    help=cli_help["tolerance_file_name"],
+    "--tolerance-files",
+    type=CommaSeparatedStrings(),
+    help=cli_help["tolerance_files_input"],
+    default=None,
 )
+@click.option("--factor", type=float, help=cli_help["factor"])
 @click.option(
-    "--factor",
-    type=float,
-    help=cli_help["factor"],
+    "--fof-types",
+    type=CommaSeparatedStrings(),
+    default="",
+    help=cli_help["fof_types"],
 )
-def check(input_file_ref, input_file_cur, tolerance_file_name, factor):
-
-    out, err, tol = check_stats_file_with_tolerances(
-        tolerance_file_name, input_file_ref, input_file_cur, factor
-    )
-
-    if out:
-        logger.info("RESULT: check PASSED!")
-    else:
-        logger.info("RESULT: check FAILED")
-        logger.info("Differences")
-        logger.info(err)
-        logger.info("\nTolerance")
-        logger.info(tol)
-        logger.info("\nError relative to tolerance")
-        logger.info(compute_division(err, tol))
-
-    sys.exit(0 if out else 1)
+@click.option("--rules", default="")
+def check(reference_files, current_files, tolerance_files, factor, fof_types, rules):
+
+    zipped = zip(reference_files, current_files, tolerance_files)
+
+    expanded_zip = expand_zip(zipped, fof_types)
+
+    all_out = True
+
+    for reference_file, current_file, tolerance_file in expanded_zip:
+
+        out, err, tol = check_file_with_tolerances(
+            tolerance_file, reference_file, current_file, factor, rules
```

Comment:
That should save some casts in `check_file_with_tolerance` and should make the code a bit cleaner.
```suggestion
            tolerance_file, FileInfo(reference_file), FileInfo(current_file), factor, rules
```